OpenRA.Mods.Common/Activities/Air/TakeOff.cs

```diff
@@ -28,7 +28,7 @@ protected override void OnFirstRun(Actor self)
 			if (aircraft.ForceLanding)
 				return;
 
-			if (self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition).Length >= aircraft.Info.MinAirborneAltitude)
+			if (!aircraft.HasInfluence())
```

Comment:
`AddInfluence` is only ever called from the `Land` activity. Won't this mean that pre-placed actors for example won't play takeoff sounds etc.?

Reply:
Note that the actor is added directly to the actor map when spawning, not via `Aircraft.AddInfluence`.

Reply:
That's an excellent catch. It raises the question of what to do if `Aircraft.landingCells` has not yet been set since that gets skipped over if the actor map is set directly.

It looks like the `OccupiedCells` method works around this by checking if the aircraft is at ground level and reporting the aircraft location, ignoring the landing cells.

I've updated `HasInfluence` to also report true if the aircraft is at ground level. I think that should solve the case for pre-placed aircraft, whilst the other check will catch the "landed but not at ground level" case from the original issue.

Reply:
Hm, `self.IsAtGroundLevel()` checks if the actor is at 0 height. What happens when the actor is at a height greater than 0 but less than `AircraftInfo.MinAirborneAltitude`?

Reply:
Then that wouldn't be considered "landed" here I suppose.

We could adjust the check in `HasInfluence` to test against `MinAirborneAltitude`. On its own though that would mean `TakeOff` would try and remove influence, but `OccupiedCells` would report no influence and therefore I imagine pre-placed aircraft above 0 but below min airborne altitude would fail to remove their influence.

So I think both `HasInfluence` and `OccupiedCells` would need to be kept in sync. Do you think the min airborne check would be better than the ground level check?

Reply:
Have updated to check against the min airborne altitude since that seemed the more sensible of the two options to me.

Reply:
That looks reasonable to me.